src/components/AlertRuleForm.tsx

```diff
@@ -0,0 +1,312 @@
+    timeCount: parseInt(timeCount, 10),
+    timeUnit: timeOption ?? {},
+    sensitivity: sensitivityOption,
+    annotations: Object.keys(rule.annotations ?? {}).map(annotationName => ({
+      name: annotationName,
+      value: rule.annotations?.[annotationName] ?? '',
+    })),
+    labels,
+  };
+};
+
+const getStyles = (theme: GrafanaTheme) => ({
+  container: css`
+    background-color: ${theme.colors.bg2};
+    padding: ${theme.spacing.md};
+    margin-bottom: ${theme.spacing.sm};
+  `,
+  inlineText: css`
+    white-space: nowrap;
+    flex-grow: 1;
+  `,
+  numberInput: css`
+    width: 75px;
+    flex-shrink: 0;
+  `,
+  selectInput: css`
+    width: 100px;
+    flex-grow: 0;
+    margin-bottom: 0;
+  `,
+  breakLine: css`
+    margin-top: 0;
+  `,
+  noMargin: css`
+    margin-bottom: 0;
+  `,
+  expressionContainer: css`
+    margin-bottom: ${theme.spacing.md};
+  `,
+  submitFail: css`
+    margin-top: ${theme.spacing.md};
+  `,
+  link: css`
+    text-decoration: underline;
+  `,
+  previewHelpText: css`
+    margin-top: ${theme.spacing.sm};
+  `,
+  preview: css`
+    border: 1px solid ${theme.colors.panelBorder};
+    background-color: ${theme.colors.panelBg};
+    margin-bottom: ${theme.spacing.sm};
+    color: ${theme.colors.textWeak};
+    line-height: 20px;
+    padding: ${theme.spacing.sm};
+  `,
+  indent: css`
+    margin-left: ${theme.spacing.sm};
+  `,
+  button: css`
+    color: ${theme.colors.textHeading};
+    background-color: ${theme.colors.bg2};
+    width: 100%;
+    border: none;
+    text-align: left;
+    padding: ${theme.spacing.md};
+    margin-bottom: ${theme.spacing.sm};
+  `,
+});
+
+type Props = {
+  rule: AlertRule;
+  onSubmit: (alertValues: AlertFormValues) => Promise<FetchResponse<unknown>>;
+};
+
+export const AlertRuleForm: FC<Props> = ({ rule, onSubmit }) => {
+  const defaultValues = getAlertFormValues(rule);
+  const { instance } = useContext(InstanceContext);
+  const styles = useStyles(getStyles);
+  const [isOpen, setIsOpen] = useState(false);
+  const formMethods = useForm<AlertFormValues>({
+    defaultValues,
+  });
+  const { register, control, handleSubmit, errors, watch, reset } = formMethods;
+  const currentValues = watch() as AlertFormValues;
+  const currentLabels = watch('labels');
+  const currentAnnotations = watch('annotations');
+
+  const onCancel = () => {
+    reset();
+    setIsOpen(false);
+  };
+
+  const { execute, error, loading: submitting } = useAsyncCallback(async (alertValues: AlertFormValues) => {
+    try {
+      await onSubmit(alertValues);
+      appEvents.emit(AppEvents.alertSuccess, ['Alert rule updated successfully']);
+      formMethods.reset(alertValues);
+      setIsOpen(false);
+      return Promise.resolve();
+    } catch (e) {
+      return Promise.reject(e.message ?? 'Something went wrong');
+    }
+  });
+
+  if (!defaultValues) {
+    return (
+      <Collapse label={rule.alert} isOpen={isOpen} onToggle={() => setIsOpen(!isOpen)} collapsible>
```

Comment:
We're no longer using a <Collapse/> for the form, instead replacing it with a button. This should match

Reply:
Yes! Thanks a bunch